tests/Microsoft.Identity.Test.Unit/PublicApiTests/DstsTests.cs

```diff
@@ -0,0 +1,97 @@
+        public override void TestInitialize()
+        {
+            base.TestInitialize();
+        }
+
+        private static MockHttpMessageHandler CreateTokenResponseHttpHandler(string authority)
+        {
+            IDictionary<string, string> expectedRequestBody = new Dictionary<string, string>();
+            expectedRequestBody.Add("scope", TestConstants.ScopeStr);
+            expectedRequestBody.Add("grant_type", "client_credentials");
+            expectedRequestBody.Add("client_id", TestConstants.ClientId);
+            expectedRequestBody.Add("client_secret", TestConstants.ClientSecret);
+
+            return new MockHttpMessageHandler()
+            {
+                ExpectedUrl = $"{authority}/dstsv2/token",
+                ExpectedMethod = HttpMethod.Post,
+                ExpectedPostData = expectedRequestBody,
+                ResponseMessage = MockHelpers.CreateSuccessfulClientCredentialTokenResponseMessage(MockHelpers.CreateClientInfo(TestConstants.Uid, TestConstants.Utid))
+            };
+        }
+
+        [DataTestMethod]
+        [DataRow("common")]
+        [DataRow("tenantid")]
+        public async Task DstsClientCredentialSuccessfulTestAsync(string tenantId)
+        {
+            string authority = $"https://foo.bar.test.core.azure-test.net/dstsv2/{tenantId}";
+
+            using (var httpManager = new MockHttpManager())
+            {
+                IConfidentialClientApplication app = ConfidentialClientApplicationBuilder
+                    .Create(TestConstants.ClientId)
+                    .WithHttpManager(httpManager)
+                    .WithAuthority(authority)
+                    .WithClientSecret(TestConstants.ClientSecret)
+                    .Build();
+
+                Assert.AreEqual(authority + "/", app.Authority);
+                var confidentailClientApp = (ConfidentialClientApplication)app;
+                Assert.AreEqual(AuthorityType.Dsts, confidentailClientApp.AuthorityInfo.AuthorityType);
+
+                httpManager.AddMockHandler(CreateTokenResponseHttpHandler(authority));
+
+                AuthenticationResult result = await app
+                    .AcquireTokenForClient(TestConstants.s_scope)
+                    .ExecuteAsync(CancellationToken.None)
+                    .ConfigureAwait(false);
+
+                Assert.IsNotNull(result);
+                Assert.IsNotNull(result.AccessToken);
+                Assert.AreEqual(TokenSource.IdentityProvider, result.AuthenticationResultMetadata.TokenSource);
+
+                result = await app
+                    .AcquireTokenForClient(TestConstants.s_scope)
+                    .ExecuteAsync(CancellationToken.None)
+                    .ConfigureAwait(false);
+
+                Assert.IsNotNull(result);
+                Assert.IsNotNull(result.AccessToken);
+                Assert.AreEqual(TokenSource.Cache, result.AuthenticationResultMetadata.TokenSource);    
+                //request to token endpoint
```

Comment:
This comment can be removed. I added this as reminder to assert it and added the checks above at line 47 but forgot to remove this.